Lib/cu2qu/ufo.py

```diff
@@ -0,0 +1,198 @@
+            pen.qCurveTo(*args[1:])
+        elif tag == 'close':
+            pen.closePath()
+        elif tag == 'end':
+            pen.endPath()
+        elif tag == 'component':
+            pen.addComponent(*args)
+        else:
+            raise AssertionError('Unhandled segment type "%s"' % tag)
+
+
+def _segments_to_quadratic(segments, max_err, stats):
+    """Return quadratic approximations of cubic segments."""
+
+    assert all(s[0] == 'curve' for s in segments), 'Non-cubic given to convert'
+
+    new_points, _ = curves_to_quadratic([s[1] for s in segments], max_err)
+    n = len(new_points[0])
+    assert all(len(s) == n for s in new_points[1:]), 'Converted incompatibly'
+
+    n = str(n)
+    if stats is not None:
+        stats[n] = stats.get(n, 0) + 1
+
+    return [('qcurve', p) for p in new_points]
+
+
+def _fonts_to_quadratic(fonts, max_err, stats):
+    """Do the actual conversion of fonts, after arguments have been set up."""
+
+    for glyphs in zip(*fonts):
+        name = glyphs[0].name
+        assert all(g.name == name for g in glyphs), 'Incompatible fonts'
+
+        segments_by_location = zip(*[_get_segments(g) for g in glyphs])
+        if not any(segments_by_location):
+            continue
+
+        new_segments_by_location = []
+        for segments in segments_by_location:
+            tag = segments[0][0]
+            assert all(s[0] == tag for s in segments[1:]), (
+                'Incompatible glyphs "%s"' % name)
+            if tag == 'curve':
+                segments = _segments_to_quadratic(segments, max_err, stats)
+            new_segments_by_location.append(segments)
+
+        new_segments_by_glyph = zip(*new_segments_by_location)
+        for glyph, new_segments in zip(glyphs, new_segments_by_glyph):
+            _set_segments(glyph, new_segments)
+
+
+def fonts_to_quadratic(fonts, max_err_em=None, max_err=None,
+        stats=None, dump_stats=False):
+    """Convert the curves of a collection of fonts to quadratic.
+
+    All curves will be converted to quadratic at once, ensuring interpolation
+    compatibility. If this is not required, calling fonts_to_quadratic with one
+    font at a time may yield slightly more optimized results.
+    """
+
+    if stats is None:
+        stats = {}
+
+    if max_err_em and max_err:
+        raise TypeError('Only one of max_err and max_err_em can be specified.')
+    if not (max_err_em or max_err):
+        max_err_em = DEFAULT_MAX_ERR
+
+    if isinstance(max_err, (list, tuple)):
+        max_errors = max_err
+    elif isinstance(max_err_em, (list, tuple)):
+        max_errors = max_err_em
+    elif max_err:
+        max_errors = [max_err] * len(fonts)
+    else:
+        max_errors = [f.info.unitsPerEm * max_err_em for f in fonts]
+
+    num_fonts = len(fonts)
+    assert len(max_errors) == num_fonts
+
+    _fonts_to_quadratic(fonts, max_errors, stats)
+
+    if dump_stats:
+        spline_lengths = stats.keys()
+        spline_lengths.sort()
```

Comment:
python3's `dict.keys()` does not return a list but a `set`-like object which doesn't have a `sort()` method.
You should use `sorted(spline_lengths.keys())` if you want to get a sorted list of keys in both python2 and 3.